### PR TITLE
Add double check for pipeline update if pipeline exists

### LINKED ISSUE #456: Need double check when updating the pipeline if already exist
Currently, when running `archer pipeline update`, it either deploy a new pipeline or update the current pipeline. However, this can cause a problem for those users who accidentally run the command and do not mean to update their pipeline (since the `update` command contains both "create" and "update" functionalities). To fix this issue we need to add double check prompt to make sure users really want to update their pipeline.

### PR DESCRIPTION
**Pull request by iamhopaul123**
Fix #456.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: internal/pkg/cli/deploy.go [hunk 1]
```diff
@@ -16,7 +16,9 @@ type environmentDeployer interface {
 }
 
 type pipelineDeployer interface {
-	DeployPipeline(env *deploy.CreatePipelineInput) error
+	CreatePipeline(env *deploy.CreatePipelineInput) error
+	UpdatePipeline(env *deploy.CreatePipelineInput) error
+	PipelineExist(env *deploy.CreatePipelineInput) (bool, error)
```
Comment (efekarakus) [inline internal/pkg/cli/deploy.go]: nit: Exist**s**

File: internal/pkg/cli/pipeline_update.go [hunk 2]
```diff
@@ -118,6 +126,51 @@ func (opts *UpdatePipelineOpts) getArtifactBuckets() ([]deploy.ArtifactBucket, e
...
+	if !exist {
+		opts.prog.Start(fmt.Sprintf(fmtCreatePipelineStart, color.HighlightUserInput(opts.PipelineName)))
+		err := opts.pipelineDeployer.CreatePipeline(in)
+		var alreadyExists *cloudformation.ErrStackAlreadyExists
+		if err == nil || errors.As(err, &alreadyExists) {
+			opts.prog.Stop(log.Ssuccessf(fmtCreatePipelineComplete, color.HighlightUserInput(opts.PipelineName)))
+			return nil
+		}
+		opts.prog.Stop(log.Serrorf(fmtCreatePipelineFailed, color.HighlightUserInput(opts.PipelineName)))
+		return fmt.Errorf("create pipeline: %w", err)
```
Comment (efekarakus) [inline internal/pkg/cli/pipeline_update.go]: nit: What do you think about moving the failure case to the top?
```go
if err := opts.pipelineDeployer.CreatePipeline(in); err != nil {
  var alreadyExists *cloudformation.ErrStackAlreadyExists
  if !errors.As(err, &alreadyExists) {
     opts.prog.Stop(log.Serrorf(fmtCreatePipelineFailed, color.HighlightUserInput(opts.PipelineName)))
     return fmt.Errorf("create pipeline: %w", err)
  }
}
// success case
```

File: internal/pkg/cli/pipeline_update.go [hunk 3]
```diff
@@ -118,6 +126,51 @@ func (opts *UpdatePipelineOpts) getArtifactBuckets() ([]deploy.ArtifactBucket, e
...
+	}
+
+	// If the stack already exists - we update it
+	shouldUpdate, err := opts.shouldUpdate()
+	if err != nil {
+		return err
+	}
+	if shouldUpdate {
```
Comment (efekarakus) [inline internal/pkg/cli/pipeline_update.go]: We can flip the if-statement to reduce nesting:
```go
if !shouldUpdate {
  return nil
}
opts.prog.Start.....
```

File: internal/pkg/deploy/cloudformation/pipeline.go [hunk 4]
```diff
@@ -16,11 +16,23 @@ import (
...
+func (cf CloudFormation) PipelineExist(in *deploy.CreatePipelineInput) (bool, error) {
+	stackConfig := stack.NewPipelineStackConfig(in)
+	describeStackInput := &cloudformation.DescribeStacksInput{StackName: aws.String(stackConfig.StackName())}
+	_, err := cf.describeStack(describeStackInput)
+	if err != nil {
+		var stackNotFound *ErrStackNotFound
+		if !errors.As(err, &stackNotFound) {
+			return false, err
+		}
+		return false, nil
```
Comment (efekarakus) [inline internal/pkg/deploy/cloudformation/pipeline.go]: nice 😁

File: internal/pkg/cli/pipeline_update.go (hunk 2 above)
Comment (iamhopaul123) [reply internal/pkg/cli/pipeline_update.go]: That's much better :+1:

**Pull request merged**